src/istio/control/tcp/attributes_builder_test.cc

```diff
@@ -73,16 +74,22 @@ attributes {
     string_value: "www.google.com"
   }
 }
+attributes {
+  key: "source.namespace"
+  value {
+    string_value: "test_ns"
+  }
+}
 attributes {
   key: "source.principal"
   value {
-    string_value: "test_user"
+    string_value: "cluster.local/sa/test_user/ns/test_ns/"
```

Comment:
[nit] Add test cases with having extra "ns/" or "/ns" to be sure.

Reply:
Updated in both attributes_builder_test.cc and utils_test.cc